packages/fxa-settings/src/pages/ResetPassword/AccountRecoveryResetPassword/index.tsx

```diff
@@ -287,15 +295,25 @@ const AccountRecoveryResetPassword = ({
           );
           break;
         case IntegrationType.OAuth:
-          if (
-            sessionIsVerified &&
-            // a user can only redirect back to the relier from the original tab
-            // to avoid two tabs redirecting.
-            isOriginalTab()
-          ) {
-            // TODO: this.finishOAuthSignInFlow(account)) in FXA-6518 and possibly
-            // remove the !OAuth check from the React experiment in router.js
-            return;
+          if (sessionIsVerified) {
+            const oauthIntegration = integration as OAuthIntegration;
+            const { redirect } = await oauthIntegration.handlePasswordReset(
+              relier.uid || account.uid,
+              accountResetData.sessionToken,
+              accountResetData.keyFetchToken,
+              accountResetData.unwrapBKey
+            );
+
+            // Clear session / local storage states
+            clearOAuthData();
+
+            // If the user is on a single tab throughout this process, just redirect them
+            // back to the relying party. Otherwise, show them a success message
+            if (isOriginalTab()) {
+              clearOriginalTab();
+              window.location.href = redirect;
```

Comment:
I wonder if we might want a `hardNavigate` function for this to make it more obvious that it's intentionally navigating away from our React app, and then update `hardNavigateToContentServer` to call it.